### PR TITLE
Fix list job -x

### PR DESCRIPTION
**Pull request by nitinjainsj**
[ This is a placeholder PR for discussion. We need to discuss `list job -x` behavior wrt to history ]

Looking at the output of `list job` and `list job -x` after an updated pipeline showed a different list. Turns out history is set to none for `-x`

```
$ pc list job 
ID                               SUBJOBS PROGRESS CREATED      MODIFIED
b7206e1f14e743b790c91797c8128394 1       ▇▇▇▇▇▇▇▇ 27 hours ago 27 hours ago 
d2e7ac68e8134671abfa907dd9784b76 2       ▇▇▇▇▇▇▇▇ 27 hours ago 27 hours ago 
31dfe367cc6448c390ea45cd155efc1a 2       ▇▇▇▇▇▇▇▇ 27 hours ago 27 hours ago 
9d5a96e210ae47f3bd1edbf4709cb851 1       ▇▇▇▇▇▇▇▇ 27 hours ago 27 hours ago 
45fe7aa3fdb840cabbade49775f6eede 1       ▇▇▇▇▇▇▇▇ 27 hours ago 27 hours ago
$ pc list job -x 
ID                               PIPELINE STARTED      DURATION           RESTART PROGRESS  DL       UL       STATE   
b7206e1f14e743b790c91797c8128394 montage  27 hours ago 4 seconds          0       1 + 0 / 1 371.9KiB 1.292MiB success 
d2e7ac68e8134671abfa907dd9784b76 edges    27 hours ago 6 seconds          0       2 + 1 / 3 181.1KiB 111.4KiB success 
31dfe367cc6448c390ea45cd155efc1a edges    27 hours ago 2 seconds          0       1 + 0 / 1 57.27KiB 22.22KiB success 
45fe7aa3fdb840cabbade49775f6eede edges    27 hours ago Less than a second 0       0 + 0 / 0 0B       0B       success
```

As you can see, there are jobs missing in `list job -x` that are listed in `list job`

The output should be????
```
$ pc list job -x
ID                               PIPELINE STARTED      DURATION           RESTART PROGRESS  DL       UL       STATE   
b7206e1f14e743b790c91797c8128394 montage  27 hours ago 4 seconds          0       1 + 0 / 1 371.9KiB 1.292MiB success 
d2e7ac68e8134671abfa907dd9784b76 montage  27 hours ago 5 seconds          0       1 + 0 / 1 371.9KiB 1.292MiB success 
d2e7ac68e8134671abfa907dd9784b76 edges    27 hours ago 6 seconds          0       2 + 1 / 3 181.1KiB 111.4KiB success 
31dfe367cc6448c390ea45cd155efc1a edges    27 hours ago 2 seconds          0       1 + 0 / 1 57.27KiB 22.22KiB success 
31dfe367cc6448c390ea45cd155efc1a montage  27 hours ago 2 seconds          0       1 + 0 / 1 79.49KiB 381.1KiB success 
9d5a96e210ae47f3bd1edbf4709cb851 montage  27 hours ago Less than a second 0       0 + 0 / 0 0B       0B       success 
45fe7aa3fdb840cabbade49775f6eede edges    27 hours ago Less than a second 0       0 + 0 / 0 0B       0B       success
```

The change here is to default history to all for list job --> I think this is the intended behavior of `list job -x` which is to expand on the `list job` output.

Side note... since history is used in multiple commands the default is taken from the last executed Flag statement. In this case it was in `list pipeline`. Hence the code change to disambiguate the defaults for both the commands.

### CODE REVIEW

File: src/server/pps/cmds/cmds.go [hunk 1]
```diff
@@ -164,7 +164,8 @@ If the job fails, the output commit will not be populated with data.`,
 
 	var pipelineName string
 	var inputCommitStrs []string
-	var history string
+	var listPipelineHistory string
+	var listJobHistory string
```
Comment (jdoliner) [inline src/server/pps/cmds/cmds.go:168]: This change is fine but shouldn't change behavior. Only one of `list pipeline` or `list job` will run per invocation of `pachctl` so the variable can't persist between invocations.

Comment (nitinjainsj) [reply src/server/pps/cmds/cmds.go:168]: Yep. The issue here is... trying to set different defaults for "history" in `list pipeline` and `list job` does not work if we have the same variable. The second invocation of `Flags().StringVar(&history...` sets the default value. Which in this case was from `list pipeline`

**Comment by nitinjainsj** [comment]
@Tryneus In order to make things consistent, we should have "history" support for `list job`. From the implementation, seems like we explicitly prevent that today. Keeping this open to get your thoughts and reason.

**Comment by nitinjainsj** [comment]
Stale PR

**Pull request closed without merging**